naga/src/valid/type.rs

```diff
@@ -583,6 +583,26 @@ impl super::Validator {
                                 Err((handle, Disalignment::UnsizedMember { index: i as u32 }));
                         }
                     }
+                    if member.binding.is_some()
+                        && !base_info.flags.contains(TypeFlags::IO_SHAREABLE)
+                    {
+                        match member.binding.as_ref().unwrap() {
+                            crate::Binding::BuiltIn(_) => { // do nothing
+                            }
+                            crate::Binding::Location {
+                                location: _,
+                                second_blend_source: _,
+                                interpolation: _,
+                                sampling: _,
+                            } => {
+                                let default = std::format!("{}", member.ty.index());
+                                let st = gctx.types[member.ty].name.as_ref().unwrap_or(&default);
+                                return Err(TypeError::LocationMemberUsingInvalidType {
+                                    struct_type: std::format!("[{}]", st),
+                                });
+                            }
+                        }
+                    }
```

Comment:
I don't think we need this block as we check this here:

https://github.com/gfx-rs/wgpu/blob/07e59eb6fc7de3f682f1c401b9cf9f0da9ee4b4a/naga/src/valid/interface.rs#L268-L274